### DISCUSSION TITLE
Saving a new file, cannot see previously saved filed

### DISCUSSION THREAD

**Discussion by romelussaladar** [aclap-dev/video-downloadhelper#112] [Help and Q&A] [answered]
### Your question

I am using 8.2.0.23 and every time I download a video clip from TWITCH, I noticed that I cannot see previously saved files downloaded by videodownloadhelper. For me when I save a file from TWITCH, it automatically creates a name ***Twitch.mp4***, and when I change that file name for example like ***Hanavocado SINGING 15Feb24 Twitch.mp4*** , it saves the file, but when I make another clip in twitch, and download that new file with downloadhelper, I cannot see the earlier file name, so I can't use that old file and add a number on it ***Hanavocado SINGING part2 15Feb24 Twitch.mp4*** . Basically after already downloading over 40 files, I cannot see any of the VideoDownloadHelper files at all, I have to open another window to look in the folder in the DOWNLOAD directory to see the files downloaded. I can't see any of these names in the same folder list when I save a NEW video to download. What it does is create a new file itself like ***Twitch-2.mp4***, and I have to edited that typing my own file name(or going into the windows FOLDER and into the download directory to see the list of previous files downloaded, to copy the text name of the file, then use that to replace the ***Twitch-2.mp4*** file name. 

Therefore I cannot see any downloaded file in the directory list when I click save to download a new video. In the past every time I saved a video to download, I could see the previous file in the list, then I would click on that name and add a new number, so I have multiple videos with ***Hanavocado SINGING part 3 15Feb23 Twitch.mp4***  parts 3, 4 , 5, 6 etc etc etc. NOW I cannot see these previous files at all. I can see other files ***NOT DOWNLOADED*** with videodownloadhelper.

**Comment by paulrouget** [comment]
Hey, I'll be happy to help you.

Like I said last time, ⚠️⚠️⚠️⚠️ please stop using caps ⚠️⚠️⚠️⚠️. It makes your whole text unreadable.

Also, please add the details of your setup, as explained here: https://github.com/aclap-dev/video-downloadhelper/wiki/Where-can-I-find-my-versions%3F

**Reply by romelussaladar** [reply]
<img width="956" alt="Screenshot 2024-02-16 212920" src="https://github.com/aclap-dev/video-downloadhelper/assets/111546622/5d127f8c-0d2a-4472-b579-1b4065781486">


Hello @paulrouget,

I am continuing to download using the latest chrome update, with latest videodownloadhelper 8.2.0.23 . Just to let you also know, as I continue to use the current version to download more twitch clips, all previous videodownloadhelper download files before 14Feb24 can be see in the list, when I click to save a new video file. All recent downloads with the new 8,2.0.23 I can't see in the list at all. I can only see them when going outside chrome, into windows folder and choosing the download directory. I am using windows 11 home. I attempted to paste the chrome info screen grab to this message.

**Comment by GrampaWildWilly** [comment]
I don't believe I have ever encountered the problem you are describing.  Let me try to replicate what you are doing.  I've picked a random video to download:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/267e2482-e4af-476d-9d82-5c3f236d596b)

You can see that this target directory already has some files in it, obviously not previously downloaded by VDH.

Now I'm going to download some other randomly selected video:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/c83587f0-63c0-486b-8f84-1a767253f73c)

It shows the first downloaded file here.

Is this more or less what you are saying you are NOT seeing?  Do feel free to post some screenshots of what you do see.

For completeness, let me say that I am running Windows 11 Pro 64-bit version 23H2 build 22631.3155, Firefox 122.0.1 64-bit, licensed VDH 8.2.0.23a1 beta, CoApp 2.0.17.

**Reply by romelussaladar** [reply]
Hmmm...  ok what I have attempted just now... is to save a new twitch video, keeping the name that Videodownloadhelper suggested which is ***Twitch.mp4*** . I then attempted to download a new video from twitch clips, and when I click save, it shows ***Twitch-2.mp4*** and I saved this file also to download. When I clicked on a new video to download and it showed ***Twitch-3.mp4*** in the list below showing all files, I could not see the previous 2 files downloaded a few minutes ago. I can only see these files when I go out of chrome into windows folder, and choose the download directory to see the 2 twitch video files downloaded. For some reason when I click save for subsequent new downloads, it does not show the previous downloaded files in the list underneath showing the file names.

Hope this makes sense.

**Reply by GrampaWildWilly** [reply]
I just wanted to point out the file names here:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/2bd1b753-93fe-4052-80eb-d763b4d9d3e4)

Despite the Japanese (I think) characters in the one file name & the little graphics in the other one, both downloads completed successfully & VLC played these two files with no problems, video & audio.

**Reply by GrampaWildWilly** [reply]
One or two screenshots would help it make a lot more sense.

**Reply by romelussaladar** [reply]
yeah they download completely and videos work. I can see them in the windows folder download directory. The files are good...

It;s when I am downloading the files clicking ***save*** in that section I can't see the previous latest files already downloaded. In the past before 3 days ago, I could always see all previous files videodownloadhelper downloaded(and all other files in the directory).

**Reply by GrampaWildWilly** [reply]
I know your downloads are good.  What I want to see  -- TO SEE -- is the VDH file selection dialog you are getting.  I want to SEE it.  Post screenshots.

**Reply by romelussaladar** [reply]
<img width="955" alt="Screenshot 2024-02-16 215728" src="https://github.com/aclap-dev/video-downloadhelper/assets/111546622/ea016b83-89af-4b43-b4c3-32d456921bbc">
<img width="955" alt="Screenshot 2024-02-16 215812" src="https://github.com/aclap-dev/video-downloadhelper/assets/111546622/0e922d3d-f806-4418-91bd-d764d840189b">
<img width="957" alt="Screenshot 2024-02-16 215842" src="https://github.com/aclap-dev/video-downloadhelper/assets/111546622/eae19b9b-5744-4f54-a07b-14cf18a73a6f">
<img width="958" alt="Screenshot 2024-02-16 215907" src="https://github.com/aclap-dev/video-downloadhelper/assets/111546622/c6253b11-3954-498b-82c8-f47fd118ede7">
<img width="956" alt="Screenshot 2024-02-16 215946" src="https://github.com/aclap-dev/video-downloadhelper/assets/111546622/ef2bb4e6-ccac-4cc1-8a9f-8f2b2b1e635d">

**Reply by romelussaladar** [reply]
ok...

when I create a new folder inside the download folder, called ***test download***, inside that folder I can see all the new videos downloaded in there. When I use the original directory called ***download*** (which is a windows created default folder for downloads)  I cannot see the files previously downloaded in there. I replicated what you did above creating a new ***test download*** directory, and yes I can see the files in there.. I created the ***testing download*** directory as a sub folder inside the ***download folder***.

**Reply by romelussaladar** [reply]
Inside the sub directory called ***test download***, when I click save for another clip for VDH to get, I can see all previous file names in the LIST below.

When I use the previous ***download*** folder (a windows created download directory) , for some reason I can't see any of the previous files, the same way I can see in the ***test download*** sub directory.

Hope this makes sense.

**Reply by romelussaladar** [reply]
@GrampaWildWilly  just to be clear, the issue is when I click ***save*** in VDH to download a file, and in my original ***download*** folder, I could never see any of the new videos downloaded by 8.2.0.23 . When I click ***save*** and use the sub folder ***test download*** , then I can see in the list all previous files downloaded by VDH (just before I click save). All the images earlier demonstrated that.

**Reply by paulrouget** [reply]
@GrampaWildWilly do you think it could be related to https://github.com/aclap-dev/video-downloadhelper/discussions/92 ?

**Reply by GrampaWildWilly** [reply]
I didn't 100% follow that other one because it is about Linux, a system I know nothing about.  But it does seem like there are some similarities.  The other user is not seeing the full range of the folders on a partition.  Our user here is not seeing previous files in a given directory.  But it seems specific to the Windows system default download directory.  I never use that directory myself, even when I do non-VDH downloads using Firefox.  I hate the whole philosophy of mixing user files into the operating system's directory tree.  That seems to be the programming philosophy Microsoft has always espoused.  I'm not a Windows programmer but if I were, I would make my application install in its own directory at the root of C: or any other partition the user selected.  All of my program's files would go into some directory tree under my application's root directory.  Once the user started executing my program, all of its data would similarly go into my application's directory tree, or some place I would let my user specify.  In particular, any configuration settings for my application would go into that same directory tree.  I would not scatter my stuff to C:\Documents and Settings, C:\Users, C\Program Files, or anything else Microsoft defined during system installation.  But that's just me ranting about programming philosophy.  The problem here seems to be something weird about the system default download directory.  And it can't be something to do with permissions because our user here created a subdirectory inside there & that works the way it's supposed to, apparently.  Let me do some experimenting . . .

**Reply by GrampaWildWilly** [reply]
When I started, my Downloads directory was empty.  I put 2 meaningless files into it to start with.  Then I downloaded the same files again:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/377539e5-8fb3-4369-8ea7-4e3d629a079a)

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/e761a7c3-ae2b-4cb7-9d15-2292a2de1e3a)

I'm not getting the same effects as our user here.  This is most strange.

I'm also reluctant to draw a lot of parallels with the other problem because here, we're dealing with Windows, there it's Linux.  Different file systems entirely.

**Comment by GrampaWildWilly** [comment]
@romelussaladar What do you get when you look in the Properties dialog on the Security tab on your Downloads directory?  Here's what I have, for comparison.

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/e2b0b9cf-fc23-4c94-b880-83982ffda060)

**Reply by romelussaladar** [reply]
<img width="956" alt="Screenshot 2024-02-16 230900" src="https://github.com/aclap-dev/video-downloadhelper/assets/111546622/e0cfb690-ba29-46fa-8e30-ec3ffe774ede">

THis is mine... doesn't appear any different for the most part I think ...

**Reply by GrampaWildWilly** [reply]
Yeah.  Looks the same.  Not similar.  Identical.

Oh.  What about this?

```
C:\Users\WildWilly2>attrib Downloads
     R               C:\Users\WildWilly2\Downloads
```
Now that's a surprise.  This directory is read-only.  But but but . . .  I just copied 2 files into this directory, then downloaded 2 files into this directory.  Doesn't read-only mean anything?  This is getting weirder by the minute.

**Reply by romelussaladar** [reply]
Just to let you guys know... @GrampaWildWilly @paulrouget 

When I am watching a current livestream on twitch, I can download it from the time I click download, and it records from that point. When I stop that recording download, it saves in the dwhelper directory. All livestream recordings work as normal, always recorded in the dwhelper folder. When I click ***save*** and choose into the dwhelper directory, I can see the list of all previous files, including the latest vdh downloads I tested in the last 10 minutes.

When I attempted to choose the ***download*** directory, and save several twitch clips, I can never see the previous files in the list, each time I save a new VDH video clip, doesnt matter from twitch or other streaming site. When I chose the subfolder ***TEST DOWNLOAD*** , for each ***save*** for each VDH video to grab from twitch, I can see all the previous and latest videos recently downloaded. 

The issue is with the windows ***download*** directory.  My work around for now, will be to use the subfolder ***test directory*** or the ***dwhelper*** directory for all future downloads. 

Hope this helps a little.

**Reply by romelussaladar** [reply]
<img width="442" alt="Screenshot 2024-02-16 233254" src="https://github.com/aclap-dev/video-downloadhelper/assets/111546622/c9ffb462-cb7e-4e74-a428-bb5b1a4b4f49">


OK yeah when I go into CMD and do the attrib for downloads folder.. it does say ***R***  also...

**Reply by GrampaWildWilly** [reply]
Save?  I don't know what you are talking about.  I know Download.  But Save?  Where do you see a Save function?  Do show me a screenshot.

**Reply by GrampaWildWilly** [reply]
Wait.  That's stupid.  You're talking about the file selection dialog.  It says "Save file as..." at the top of the dialog.  So are you talking about the difference between a Quick Download & a Download?

**Reply by romelussaladar** [reply]
<img width="957" alt="Screenshot 2024-02-16 215842" src="https://github.com/aclap-dev/video-downloadhelper/assets/111546622/1ad9a9fb-d5b1-4de7-8368-4de2b4776d91">
......
<img width="958" alt="Screenshot 2024-02-16 215907" src="https://github.com/aclap-dev/video-downloadhelper/assets/111546622/850e167a-f328-48ef-b3a7-dfe12b5b5803">
---

When I choose in VDH to ***download*** it goes into a new ***save*** section, and in here I type the name of the file, and at the bottom right side you can see ***cancel*** and ***save*** . This is when I save the file in the directory I selected.

**Reply by GrampaWildWilly** [reply]
Right.  And you're saying when your target directory is C:\Users\...\Downloads, it doesn't show you the full contents of the directory.  But for any other directory, it shows you the full contents.  Have I got it?

**Reply by GrampaWildWilly** [reply]
I'm picturing Paul sitting at home reading this . . .

I've got 2 users, both using Windows, both trying to save into the same directory on their respective systems.  One is seeing anomalous behavior, the other is seeing no problems.  How am I supposed to write code for these guys?

**Comment by romelussaladar** [comment]
> Right. And you're saying when your target directory is C:\Users...\Download, it doesn't show you the full contents of the directory. But for any other directory, it shows you the full contents. Have I got it?

***Yes***

**Reply by GrampaWildWilly** [reply]
What Windows are you running?  I said above what I'm running.  What are you running?

I think at this point, your system is trying to tell you something.  I already said what I think about the idea of writing user data into any part of the directory structure Windows builds during installation.  I think your system is trying to tell you the same thing.  Download elsewhere.  I have a directory I normally use for testing things for users in this forum.  You can see it above.  It's M:\VDH Testing.  It's not even on the boot partition.  It's physically on another hard drive.  I have a number of directories into which I download stuff regularly.  None of them is on C:.  If you are going to target downloads to your C: drive, I would recommend you not put them in any directory constructed by the Windows installation.  It's just my philosophy.  I don't like the idea of mixing any of my data in with the operating system.  It looks like following that approach would clear things up for you.

**Reply by GrampaWildWilly** [reply]
Oh.  Here's some good news.  Look over here: #92.  Paul just updated the beta.  Look at his comment in there from just a few minutes ago.  I'll bet that clears this right up in no time.

**Reply by romelussaladar** [reply]
Aaah ok ok... cool... good to see how the feedback can help create the changes..  Thanks very much @GrampaWildWilly @paulrouget  ..

**Reply by paulrouget** [reply, marked as answer]
Waiting for Google to validate the beta. Hope this will improve the situation.

**Comment by juddiDaniel** [comment]
For quick and safe Twitch downloads use [ZeTwitch](https://zetwitch.com/), I have been using it and I am recommending it.
